src/Ocelot/Errors/QoS/RequestTimedOutError.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace Ocelot.Errors.QoS;
+
+public class RequestTimedOutError : Error
+{
+    public RequestTimedOutError(Exception exception)
+        : base($"Timeout making http request, exception: {exception}", OcelotErrorCode.RequestTimedOutError, 503)
```

Comment:
Do we override 408 RequestTimeout here?

Reply:
good point... I just moved the Error class from the polly provider to the base project. I don't know, what should we return, 408 or 503?

Reply:
@raman-m Should we keep it like that for the time being and create a new issue later?

Reply:
Returning **5xx** status is very bad in my opinion.
In addition we should think in terms of returning correct native status code from downstream, at least try to forward this native status.
But this is QoS logic where we try to override behavior of internal Polly features.
If it is possible it is better to return original status.
My concern was/is Why custom Error object? Cannot we just return exception or real result object.
Or this Error object plays the role of the result?